crates/optimism/txpool/src/transaction.rs

```diff
@@ -91,12 +93,12 @@ impl<Cons, Pooled> MaybeConditionalTransaction for OpPooledTransaction<Cons, Poo
 }
 
 impl<Cons, Pooled> MaybeInteropTransaction for OpPooledTransaction<Cons, Pooled> {
-    fn set_interop(&self, interop: TransactionInterop) {
-        *self.interop.write() = Some(interop);
+    fn set_interop(&self, interop: u64) {
+        self.interop.store(interop, Ordering::Relaxed);
     }
 
-    fn interop(&self) -> Option<TransactionInterop> {
-        self.interop.read().clone()
+    fn interop(&self) -> u64 {
```

Comment:
this should return an Option, and check if the stored value is != the `NO_INTEROP` const